elastic/kibana
# [ILM] Add support for frozen phase in UI

## Summary

Support configuring the frozen phase in both server and UI of ILM. This also updates the index management plugins extension to support querying for "frozen" phase in the index table.

## How to test

1. Create a new ILM policy that consists of Hot and Frozen phases only (to simplify testing). In the UI, make sure that you use all options for frozen phase except for setting required replicas as this will cause the phase to get stuck there if the cluster cannot create a replica of the shards.
2. Follow this [tutorial](https://www.elastic.co/guide/en/elasticsearch/reference/current/getting-started-index-lifecycle-management.html) to set up an index template that will use the ILM policy and  create an initial write index (making sure to keep your naming consistent). Make sure to set any timing values to one second (`1 second`) so that you do not have to wait for phase rollover too long.
3. Run the following in Dev Tools to make ILM check progress more frequently:
```
PUT _cluster/settings
{
  "transient": {
    "indices.lifecycle.poll_interval": "5s"  
  }
}
```

Make sure to set this back to the default of `10m` so that ILM doesn't continue doing unnecessary work after testing.

What you should see in index management is new indices being created every 5 seconds as rollover happens and they should have the "Frozen" label applied to them in the index management UI.

Fix https://github.com/elastic/elasticsearch/issues/61345

## Screenshots

<img width="1203" alt="Screenshot 2020-08-26 at 12 56 47" src="https://user-images.githubusercontent.com/8155004/91295622-a573d380-e79b-11ea-8b26-03d809d7a2be.png">


### Checklist

Delete any items that are not applicable to this PR.
